src/main/java/io/airlift/slice/Slice.java

```diff
@@ -156,133 +110,22 @@ public final class Slice
-     * @param length the number of array positions to include in the slice
-     */
-    Slice(int[] base, int offset, int length)
-    {
-        requireNonNull(base, "base is null");
-        checkFromIndexSize(offset, length, base.length);
-
-        this.base = base;
-        this.address = sizeOfIntArray(offset);
-        this.size = multiplyExact(length, ARRAY_INT_INDEX_SCALE);
-        this.retainedSize = INSTANCE_SIZE + sizeOf(base);
-        this.reference = (offset == 0 && length == base.length) ? COMPACT : NOT_COMPACT;
-    }
-
-    /**
-     * Creates a slice over the specified array range.
-     *
-     * @param offset the array position at which the slice begins
-     * @param length the number of array positions to include in the slice
-     */
-    Slice(long[] base, int offset, int length)
-    {
-        requireNonNull(base, "base is null");
-        checkFromIndexSize(offset, length, base.length);
-
-        this.base = base;
-        this.address = sizeOfLongArray(offset);
-        this.size = multiplyExact(length, ARRAY_LONG_INDEX_SCALE);
-        this.retainedSize = INSTANCE_SIZE + sizeOf(base);
-        this.reference = (offset == 0 && length == base.length) ? COMPACT : NOT_COMPACT;
-    }
-
-    /**
-     * Creates a slice over the specified array range.
-     *
-     * @param offset the array position at which the slice begins
-     * @param length the number of array positions to include in the slice
-     */
-    Slice(float[] base, int offset, int length)
-    {
-        requireNonNull(base, "base is null");
-        checkFromIndexSize(offset, length, base.length);
-
-        this.base = base;
-        this.address = sizeOfFloatArray(offset);
-        this.size = multiplyExact(length, ARRAY_FLOAT_INDEX_SCALE);
-        this.retainedSize = INSTANCE_SIZE + sizeOf(base);
-        this.reference = (offset == 0 && length == base.length) ? COMPACT : NOT_COMPACT;
-    }
-
-    /**
-     * Creates a slice over the specified array range.
-     *
-     * @param offset the array position at which the slice begins
-     * @param length the number of array positions to include in the slice
-     */
-    Slice(double[] base, int offset, int length)
-    {
-        requireNonNull(base, "base is null");
-        checkFromIndexSize(offset, length, base.length);
-
-        this.base = base;
-        this.address = sizeOfDoubleArray(offset);
-        this.size = multiplyExact(length, ARRAY_DOUBLE_INDEX_SCALE);
-        this.retainedSize = INSTANCE_SIZE + sizeOf(base);
-        this.reference = (offset == 0 && length == base.length) ? COMPACT : NOT_COMPACT;
     }
 
     /**
      * Creates a slice for directly accessing the base object.
      */
-    Slice(@Nullable Object base, long address, int size, long retainedSize, @Nullable Object reference)
+    Slice(byte[] base, long address, int size, long retainedSize)
     {
-        if (address <= 0) {
+        if (address < ARRAY_BYTE_BASE_OFFSET) {
```

Comment:
We can also validate that `size` is within the bounds of the array

Reply:
This is added when the code is switched to byte[]